### ISSUE TITLE
Implement CNN-based Image Classification (Python + PyTorch)

### ISSUE THREAD

**Issue by 5usu** [SRA-VJTI/computer-vision-study-group#5]
## Summary
Implement a **Convolutional Neural Network (CNN)** for CIFAR-10 image classification using PyTorch.  

---

## Objective
- Understand convolutional feature hierarchies
- Implement CNN architecture with modern techniques
- Explore data augmentation and learning rate scheduling
- Visualize intermediate feature maps

---

## Tasks
- [ ] Create a new folder: `cnn_classification/`
- [ ] Implement:
  - `dataset.py` — CIFAR-10 loading, transforms (flip, crop, normalization)
  - `model.py` — Define CNN (Conv-BN-ReLU blocks + dropout)
  - `train.py` — Training loop with optimizer & LR scheduler
  - `eval.py` — Evaluate model, plot accuracy/loss
  - `visualize_features.py` — visualize filters & activations
- [ ] Train on CIFAR-10 with augmentation
- [ ] Log training progress (use TensorBoard or matplotlib)
- [ ] Save model checkpoints

---

## Directory Structure
cv_learn/
cnn_classification/
dataset.py
model.py
train.py
eval.py
visualize_features.py
README.md

---

##  Tools / Dependencies
- Python 3.10+
- PyTorch
- torchvision
- OpenCV
- tqdm
- matplotlib
- tensorboard (optional)

---

##  Example CLI
```bash
python cnn_classification/train.py --epochs 50 --lr 0.01 --batch-size 128 --save model_cnn.pth
python cnn_classification/eval.py --weights model_cnn.pth
python cnn_classification/visualize_features.py --weights model_cnn.pth --layer conv1
```

## Reference 
- PyTorch Docs: https://pytorch.org/docs/stable/

**Comment by gg21-prog** [comment]
@5usu I'd like to give this a go

**Comment by Coffeempty** [comment]
@5usu I am giving this a go too

**Comment by 5usu** [comment]
> [@5usu](https://github.com/5usu) I am giving this a go too

make sure you both work on diff branches @Coffeempty @gg21-prog
